AdventureWorks.BussinesLogic/Services/Sales/CustomersGetterService.cs

```diff
@@ -1,34 +1,88 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using AdventureWorks.Core.Entities.DTO;
+using AdventureWorks.Core.Entities.EF;
+using AdventureWorks.Core.Interfaces.BussinesLogic.Services.Sales;
 using AdventureWorks.Core.Interfaces.Persistance.Repositories;
-using AdventureWorks.UI.ViewEntities.Sales;
 using AdventureWorks.Core.Mappers.Sales;
-using AdventureWorks.Core.Interfaces.BussinesLogic.Services.Sales;
+using AdventureWorks.UI.ViewEntities.Sales;
+using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace AdventureWorks.BussinesLogic.Services.Sales
 {
     public class CustomersGetterService : ICustomersGetterService
     {
         private readonly ICustomersRepository _customerRepository;
+        private readonly IPeopleRepository _peopleRepository;
+        private readonly IStoreRepository _storeRepository;
 
-        public CustomersGetterService(ICustomersRepository customerRepository)
+        public CustomersGetterService(ICustomersRepository customerRepository, IPeopleRepository peopleRepository, IStoreRepository storeRepository)
         {
             if (customerRepository == null)
                 throw new ArgumentNullException(nameof(customerRepository));
+            if (peopleRepository == null)
+                throw new ArgumentException(nameof(peopleRepository));
+            if (storeRepository == null)
+                throw new ArgumentException(nameof(storeRepository));
 
             _customerRepository = customerRepository;
+            _peopleRepository = peopleRepository;
+            _storeRepository = storeRepository;
         }
 
         public CustomerViewEntity GetById(int customerId)
         {
+            CustomerDTO customerDTO = new CustomerDTO();
+
             var customer = _customerRepository.GetById(customerId);
-            return CustomersMapper.MapTo(customer);
+
+            Core.Entities.EF.Person person = null;
+            Core.Entities.EF.Store store = null;
+            if (customer.PersonId != null)
```

Comment:
Apply the same behaviour that the data access do. Allow you to control which queries to use.
With a simple view of the logic allow you to realize which object compose a CustomerDTO and allow you in a future to remove or add an unnecessary functionality.